templates.go

```diff
@@ -174,6 +182,9 @@ func loadConfig(resourcePath string) (*TemplateConfig, error) {
 				if err != nil {
 					return nil, errors.Wrapf(err, "Failed to parse resource: %s", resourcePath)
 				}
+			} else {
+				// Catch that we haven't initialised the primitive with values
+				parseResult.InitialisePrimitive = true
```

Comment:
I don't think we want this.  Not initializing the primitive is a valid choice.  The intention of the flag was to give the developer the choice.

Reply:
Resolved offline: The caller can still set a 0 valued primitive.